services/lesson.js

```diff
@@ -11,6 +13,40 @@ const lessonService = {
     )
 
     return await Lesson.create({ author, title, description, attachments: fileUrls })
+  },
+
+  updateLesson: async (id, currentUser, updateData) => {
+    const { title, description, attachments, rewriteAttachments = false } = updateData
+
+    const lesson = await Lesson.findById(id).exec()
+
+    if (!lesson) {
+      throw createError(404, DOCUMENT_NOT_FOUND(Lesson.modelName))
+    }
+
+    const author = lesson.author.toString()
+
+    if (currentUser !== author) {
+      throw createForbiddenError()
+    }
+
+    if (attachments) {
```

Comment:
maybe change this condition to: 
```javascript 
if (attachments?.length) { }
``` 
cause if here will be passed empty attachments and `rewriteAttachments` will be `true` so all attachments will be removed

Reply:
it's not a problem because removing attachments can be needed in some cases

Reply:
yeap, but i think it would be better to have attachments removing by one not all at once

Reply:
but anyway in case of for example deleting 1 of 3 attachments, other 2 will remain and will be sent with other update data, rewriteAttachments will be set in true and it will work